pennylane/fourier/utils.py

```diff
@@ -67,7 +67,9 @@ def get_spectrum(op, decimals):
         if isinstance(g, np.ndarray):
             matrix = g
         elif hasattr(g, "matrix"):
-            matrix = g.matrix
+            # hotfix: this only works because all
+            # non-numerical generators are Paulis.
+            matrix = g.compute_matrix()
```

Comment:
I want to add this comment so in the long run we find a better solution, and so developers looking for why their tests fail find it!

Reply:
@mariaschuld this will be partially addressed in #2030!